src/stratum_server.cpp

```diff
@@ -914,4 +916,84 @@ bool StratumServer::StratumClient::process_submit(rapidjson::Document& doc, uint
 	return static_cast<StratumServer*>(m_owner)->on_submit(this, id, job_id.GetString(), nonce.GetString(), result.GetString());
 }
 
+void StratumServer::api_update_local_stats()
+{
+	if (!m_pool->m_api) {
+		return;
+	}
+
+	if (!m_pool->params().m_localStats) {
+		return;
+	}
+
+	const time_t api_time_now = time(nullptr);
+
+	// Rate limit to no more than once in 60 seconds.
+	if (api_last_update_time == -1) {
+		api_last_update_time = api_time_now;
+	} else if (difftime(api_time_now, api_last_update_time) < 60) {
+		return;
+	} else {
+		api_last_update_time = api_time_now;
+	}
```

Comment:
@SChernykh please check this rate limiting implementation is okay

Reply:
Looks good, just rename it to `m_apiLastUpdateTime` like the other members of StratumServer. Also you forgot to initialize it in the constructor.